code/modules/clothing/spacesuits/hardsuit.dm

```diff
@@ -787,7 +787,25 @@
 	allowed = list(/obj/item/gun, /obj/item/ammo_box, /obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/melee/transforming/energy/sword/saber, /obj/item/restraints/handcuffs, /obj/item/tank/internals)
 	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/shielded/syndi
 	slowdown = 0
+	shield_state = "shield-red"
+	shield_on = "shield-red"
+
+/obj/item/clothing/suit/space/hardsuit/shielded/syndi/attackby(obj/item/W, mob/living/user, params)
+	if(W.tool_behaviour == TOOL_MULTITOOL)
+		if(shield_state == "shield-red")
+			shield_state = "shield-old"
+			shield_on = "shield-old"
+			to_chat(user, "<span class='warning'>You roll back the hardsuit's software, changing the shield's color!</span>")
+			user.update_inv_wear_suit()
 
+		else
+			shield_state = "shield-red"
+			shield_on = "shield-red"
+			to_chat(user, "<span class='warning'>You update the hardsuit's hardware, changing back the shield's color to red.</span>")
+			user.update_inv_wear_suit()
```

Comment:
Remove on tab here and remove the upper ``user.update_inv_wear_suit()``. Duplicate code should be avoided wherever possible even if it is just one line.

Reply:
You asked me on discord, but I'm just gonna answer here. Basically you have ``user.update_inv_wear_suit()`` twice in this proc and you could get rid of one of them if you removed a tab here and removed the one above this, you get rid of one line of duplicate code. This is really nitpicky really but just a good habit to get into.

Reply:
it is done